app/controllers/katello/api/rhsm/candlepin_proxies_controller.rb

```diff
@@ -114,7 +114,7 @@ def rhsm_index
     # Note that this request comes in as content-type 'text/plain' so that
     # tomcat won't parse the json.  Here we just pass the plain body through to candlepin
     def async_hypervisors_update
-      raw_json = request.body.string
+      raw_json = request.body.read
```

Comment:
In a similar situation, the following was used: https://github.com/Katello/katello/blob/master/app/controllers/katello/api/rhsm/candlepin_dynflow_proxy_controller.rb#L16